mypy_django_plugin/transformers/managers.py

```diff
@@ -0,0 +1,152 @@
+            if not self.defer_till_next_iteration(reason='method_node.type is None'):
+                raise new_helpers.TypeInfoNotFound(method_node.fullname)
+
+            arguments, return_type = build_unannotated_method_args(method_node)
+            add_method_to_class(
+                    ctx.api,
+                    ctx.cls,
+                    new_method_name,
+                    args=arguments,
+                    return_type=return_type,
+                    self_type=self_type)
+            return
+
+        method_type = method_node.type
+        if not isinstance(method_type, CallableType) and not self.defer_till_next_iteration(reason='method_node.type is not CallableType'):
+            raise new_helpers.TypeInfoNotFound(method_node.fullname)
+
+        arguments = []
+        bound_return_type = self.semanal_api.anal_type(
+                method_type.ret_type,
+                allow_placeholder=True)
+
+        assert bound_return_type is not None
+
+        if isinstance(bound_return_type, PlaceholderNode):
+            raise new_helpers.TypeInfoNotFound('return type '+method_node.fullname)
+
+        for arg_name, arg_type, original_argument in zip(
+                method_type.arg_names[1:],
+                method_type.arg_types[1:],
+                method_node.arguments[1:]):
+            bound_arg_type = self.semanal_api.anal_type(arg_type, allow_placeholder=True)
+            if bound_arg_type is None and not self.defer_till_next_iteration(reason='bound_arg_type is None'):
+                raise new_helpers.TypeInfoNotFound('of '+arg_name+' argument of '+method_node.fullname)
+
+            assert bound_arg_type is not None
+
+            if isinstance(bound_arg_type, PlaceholderNode) and self.defer_till_next_iteration('bound_arg_type is None'):
+                raise new_helpers.TypeInfoNotFound('of ' + arg_name + ' argument of ' + method_node.fullname)
+
+            var = Var(
+                    name=original_argument.variable.name,
+                    type=arg_type)
+            var.line = original_argument.variable.line
+            var.column = original_argument.variable.column
+            argument = Argument(
+                    variable=var,
+                    type_annotation=bound_arg_type,
+                    initializer=original_argument.initializer,
+                    kind=original_argument.kind)
+            argument.set_line(original_argument)
+            arguments.append(argument)
+
+        add_method_to_class(
+                ctx.api,
+                ctx.cls,
+                new_method_name,
+                args=arguments,
+                return_type=bound_return_type,
+                self_type=self_type)
+
+
+class ManagerFromQuerySetCallback(ManagerCallback):
+    def create_new_dynamic_class(self) -> None:
+
+        base_manager_info = self.callee.expr.node
+
+        if base_manager_info is None and not self.defer_till_next_iteration(reason='base_manager_info is None'):
```

Comment:
I don't think this `reason` is useful. `base_manager_info is None` _means_ something, it should be added as the reason. 
How could it happen that `TypeInfo` is `None` here? 

Also, let's see what type `self.callee.expr` is. It's an expression on the left side of the `a.b`. It could be `get_manager_class().from_queryset()` (which will be `CallExpr` as `self.callee.expr`, not a `RefExpr`), so there's no `node` attribute available, and so forth.
If it's not a `RefExpr`, a good idea would be to fallback to `Any` here.

Reply:
@kszmigiel I still want you to add proper error string here with an explanation of the conditions of why it could be `None`.